code/game/objects/items/devices/PDA/PDA.dm

```diff
@@ -328,10 +328,26 @@ var/global/list/obj/item/device/pda/PDAs = list()
 	name = "Nanotrasen Navy Captain PDA"
 	ownjob = "Nanotrasen Navy Captain"
 
+/obj/item/device/pda/heads/nt_captain/New()
+	..()
+	for(var/A in applications)
+		qdel(A)
+	for(var/app_type in (typesof(/datum/pda_app) - /datum/pda_app))
+		var/datum/pda_app/app = new app_type()
+		app.onInstall(src)
+
 /obj/item/device/pda/heads/nt_supreme
 	name = "Nanotrasen Supreme Commander PDA"
 	ownjob = "Nanotrasen Supreme Commander"
 
+/obj/item/device/pda/heads/nt_supreme/New()
```

Comment:
Don't copypaste these two New(), create a proc that achieves this. Also I don't understand why you are deleting all apps and then installing them again

Reply:
That was a long time ago so I'm not 100% sure but there was either an issue of duplicate apps, or something about getting rid of the apps of the parent PDA.